src/components/MediaQueryProvider/MediaQueryProvider.tsx

```diff
@@ -0,0 +1,41 @@
+import React, {useEffect, useState, useCallback} from 'react';
+import debounce from 'lodash/debounce';
+import {MediaQueryContext} from '../../utilities/media-query';
+import {navigationBarCollapsed} from '../../utilities/breakpoints';
+import {EventListener} from '../EventListener';
+
+interface Props {
+  children?: React.ReactNode;
+}
+
+export const MediaQueryProvider = function MediaQueryProvider({
```

Comment:
We check the breakpoint, when the state is initialized, after the first render and during resize.